Python/wolfssl-python-3.8.5.patch

```diff
@@ -0,0 +1,1995 @@
++                    if support.verbose:
++                        sys.stdout.write("\nSSLError is %r\n" % e)
++                except OSError as e:
++                    if e.errno != errno.ECONNRESET:
++                        raise
++                    if support.verbose:
++                        sys.stdout.write("\nsocket.error is %r\n" % e)
++                else:
++                    self.fail("Use of invalid cert should have failed!")
+             else:
+-                self.fail("Use of invalid cert should have failed!")
++                # TLS 1.3 perform client cert exchange after handshake\
++                s.connect((HOST, server.port))
++                try:
++                    s.write(b'data')
++                    s.read(4)
++                except ssl.SSLError as e:
++                    if support.verbose:
++                        sys.stdout.write("\nSSLError is %r\n" % e)
++                except OSError as e:
++                    if e.errno != errno.ECONNRESET:
++                        raise
++                    if support.verbose:
++                        sys.stdout.write("\nsocket.error is %r\n" % e)
++                else:
++                    self.fail("Use of invalid cert should have failed!")
+ 
+     def test_rude_shutdown(self):
+         """A brutal shutdown of an SSL server should raise an OSError
+@@ -3191,10 +3404,16 @@
+                 except ssl.SSLError as e:
+                     msg = 'unable to get local issuer certificate'
+                     self.assertIsInstance(e, ssl.SSLCertVerificationError)
+-                    self.assertEqual(e.verify_code, 20)
+-                    self.assertEqual(e.verify_message, msg)
+-                    self.assertIn(msg, repr(e))
+-                    self.assertIn('certificate verify failed', repr(e))
++                    if ssl.IS_WOLFSSL:
++                        self.assertEqual(e.verify_code, 29)
++                        msg = 'unable to verify the first certificate'
++                        self.assertEqual(e.verify_message, msg)
++                        self.assertIn('ASN no signer error to confirm failure', repr(e))
++                    else:
++                        self.assertEqual(e.verify_code, 20)
++                        self.assertEqual(e.verify_message, msg)
++                        self.assertIn(msg, repr(e))
++                        self.assertIn('certificate verify failed', repr(e))
+ 
+     @requires_tls_version('SSLv2')
+     def test_protocol_sslv2(self):
+@@ -3324,8 +3543,11 @@
+             try_protocol_combo(ssl.PROTOCOL_TLSv1_2, ssl.PROTOCOL_SSLv2, False)
+         if has_tls_version('SSLv3'):
+             try_protocol_combo(ssl.PROTOCOL_TLSv1_2, ssl.PROTOCOL_SSLv3, False)
+-        try_protocol_combo(ssl.PROTOCOL_TLSv1_2, ssl.PROTOCOL_TLS, False,
+-                           client_options=ssl.OP_NO_TLSv1_2)
++        
++        if not ssl.IS_WOLFSSL:
++            # this expectes to work
```

Comment:
Can you elaborate/improve this comment?

Reply:
This excluded test has been added back, changes removed from patch file. Fixed with wolfSSL PR 4525 (https://github.com/wolfSSL/wolfssl/pull/4525)